.github/scripts/close_nonexistent_disable_issues.py

```diff
@@ -0,0 +1,174 @@
+    )
+select
+    id,
+    name
+from
+    workflow_job j
+    join shas on shas.sha = j.head_sha
+where
+    j.name like '% / test%'
+    and j.name not like '%rerun_disabled_tests%'
+    and j.name not like '%mem_leak_check%'
+"""
+
+TEST_EXISTS_QUERY = """
+select
+    count(*) as c
+from
+    test_run_s3
+where
+    cast(name as string) like :name
+    and classname like :classname
+    and _event_time > CURRENT_TIMESTAMP() - DAYS(7)
+"""
+CLOSING_COMMENT = (
+    "I cannot find any mention of this test in rockset for the past 7 days "
+    "or in the logs for the past 5 commits on viable/strict.  Closing this "
+    "issue as it is highly likely that this test has either bee renamed or removed."
+)
+
+
+def parse_args() -> Any:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Only list the tests.",
+    )
+    return parser.parse_args()
+
+
+def query_rockset(
+    query: str, params: Optional[Dict[str, Any]] = None
+) -> List[Dict[str, Any]]:
+    res = rockset.RocksetClient(
+        host="api.rs2.usw2.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
+    ).sql(query, params)
+    results: List[Dict[str, Any]] = res.results
+    return results
+
+
+def download_log_worker(temp_dir: str, id: int, name: str) -> None:
+    url = f"https://ossci-raw-job-status.s3.amazonaws.com/log/{id}"
+    data = requests.get(url).text
+    with open(f"{temp_dir}/{name.replace('/', '_')} {id}.txt", "x") as f:
+        f.write(data)
+
+
+def printer(item: Tuple[str, Tuple[int, str, List[Any]]], extra: str) -> None:
+    test, (_, link, _) = item
+    print(f"{link:<55} {test:<120} {extra}")
+
+
+def close_issue(num: int) -> None:
+    headers = {
+        "Accept": "application/vnd.github.v3+json",
+        "Authorization": f"token {os.environ['GITHUB_TOKEN']}",
+    }
+    requests.post(
+        f"https://api.github.com/repos/pytorch/pytorch/issues/{num}/comments",
+        data=json.dumps({"body": CLOSING_COMMENT}),
+        headers=headers,
+    )
+    requests.patch(
+        f"https://api.github.com/repos/pytorch/pytorch/issues/{num}",
+        data=json.dumps({"state": "closed"}),
+        headers=headers,
+    )
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    disabled_tests_json = json.loads(
+        requests.get(
+            "https://raw.githubusercontent.com/pytorch/test-infra/generated-stats/stats/disabled-tests-condensed.json"
+        ).text
+    )
+
+    all_logs = []
+    jobs = query_rockset(LOGS_QUERY)
+    with tempfile.TemporaryDirectory() as temp_dir:
+        pool = mp.Pool(20)
+        for job in jobs:
+            id = job["id"]
+            name = job["name"]
+            pool.apply_async(download_log_worker, args=(temp_dir, id, name))
+        pool.close()
+        pool.join()
+
+        for filename in os.listdir(temp_dir):
+            with open(f"{temp_dir}/{filename}") as f:
+                all_logs.append(f.read())
```

Comment:
I think querying the logs from the Github API would take longer that getting them from our reupload onto AWS, so I would have to parse the step myself, so I'm not sure if this would give a lot of performance benefit.   For what it's worth, I do only look at test jobs and not build jobs.

I'm also not sure about performance since I'm not planning on running this script too often.